### PR TITLE
Add ability to hook functions with call in first 5 bytes (#4)

### PR DESCRIPTION
**Pull request by SergiusTheBest**
This is a fix for #4.

### CODE REVIEW

**Review by SergiusTheBest — COMMENTED** [review]
I think we can merge patch jump and patch call into one.

File: mhook-lib/mhook.cpp [hunk 1]
```diff
@@ -1097,6 +1096,97 @@ static bool IsJumpPresentInFirstFiveBytes(PVOID pFunction)
...
+                    // pins->Length - full legth of instruction and operand address
+                    ULONG_PTR oldStartAddress = (pLoc - pCodeTrampoline) + reinterpret_cast<ULONG_PTR>(pSystemFunction) + pins->Length;
+                    // offset from the next instruction address
+                    ULONG_PTR oldOffset = *(reinterpret_cast<UINT32*>(pins->Operands[0].BCD));
+                    // target function address
+                    ULONG_PTR destination = oldStartAddress + oldOffset;
+
+                    // now calculate new start address and new offset
+                    ULONG_PTR newStartAddress = reinterpret_cast<ULONG_PTR>(pins->Address) + pins->Length;
+                    ULONG_PTR newOffset = destination - newStartAddress;
```
Comment (SergiusTheBest) [inline mhook-lib/mhook.cpp]: `call` argument has to be 4 bytes on x64

File: mhook-lib/mhook.cpp [hunk 2]
```diff
@@ -1097,6 +1096,97 @@ static bool IsJumpPresentInFirstFiveBytes(PVOID pFunction)
...
+                    
+                    // save new offset to the trampoline code 
+                    memcpy(pLoc + pins->OpcodeLength, &newOffset, sizeof(newOffset));
```
Comment (SergiusTheBest) [inline mhook-lib/mhook.cpp]: better replace with `*reinterpret_cast<LONG*>(pLoc + pins->OpcodeLength) = newOffset`

Comment (Grivus) [reply mhook-lib/mhook.cpp]: Done

File: mhook-lib/mhook.cpp (hunk 1 above)
Comment (Grivus) [reply mhook-lib/mhook.cpp]: Done

**Review by SergiusTheBest — COMMENTED** [review]
Looks good!

**Pull request merged**